src/client.py

```diff
@@ -5,21 +5,23 @@
 import sys
 
 
-def client(message):
+def client(message, port):
     """Creates the protocol for interacting with server."""
-    info = socket.getaddrinfo('127.0.0.1', 5003)
+    if len(message) % 8 == 0:
+        message += '$'
+    info = socket.getaddrinfo('127.0.0.1', port)
     stream_info = [i for i in info if i[1] == socket.SOCK_STREAM][0]
     client_socket = socket.socket(*stream_info[:3])
     client_socket.connect(stream_info[-1])
     buffer_length = 8
     reply_complete = False
     response = ""
-    client_socket.sendall(message.encode('utf8'))
-    # client_socket.listen(3)
+    print(message.encode('utf-8'))
+    client_socket.sendall(message.encode('utf-8'))
     while not reply_complete:
-        data = client_socket.recv(buffer_length).decode('utf8')
+        data = client_socket.recv(buffer_length).decode('utf-8')
         response += data
```

Comment:
Couple of improvements to be made here.

First, in terms of efficiency, it is much better in Python to append chunks of strings to a list, then use the `str.join` method to join them all at the end, rather than concatenate one at a time like this.

Secondly, this will break sometimes on non-unicode characters. So you're decoding each chunk as you get it. Since unicode characters can take up multiple bytes, what happens when a multiple-byte unicode character isn't received all the way in one chunk? i.e. the buffer length cuts it in half?

Rather than decoding one chunk at a time, what would be a better approach?